src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "math/integerhistogram1d.h"
+#include "base/lineparser.h"
+#include "base/messenger.h"
+#include "templates/algorithms.h"
+#include <cassert>
+
+IntegerHistogram1D::IntegerHistogram1D()
+{
+    accumulatedData_.addErrors();
+
+    clear();
+}
+
+IntegerHistogram1D::IntegerHistogram1D(const IntegerHistogram1D &source) { (*this) = source; }
+
+// Clear Data
+void IntegerHistogram1D::clear()
+{
+    minimum_ = 0;
+    maximum_ = 0;
+    nBinned_ = 0;
+    nMissed_ = 0;
+    bins_.clear();
+    averages_.clear();
+}
+
+/*
+ * Data
+ */
+
+// Update accumulated data
+void IntegerHistogram1D::updateAccumulatedData()
+{
+    // Set up arrays
+    accumulatedData_.initialise(bins_.size(), true);
+
+    // Store bin centres and accumulated averages in the object
+    for (auto n = 0; n < bins_.size(); ++n)
+    {
+        accumulatedData_.xAxis(n) = n;
+        accumulatedData_.value(n) = averages_[n];
+        accumulatedData_.error(n) = averages_[n].stDev();
+    }
+}
+
+// Initialise with specified bin range
+void IntegerHistogram1D::initialise(int xMin, int xMax, int binWidth)
+{
+    clear();
+
+    // Store axis limits and binWidth, and then clamp as necessary
+    minimum_ = xMin;
+    maximum_ = xMax;
+
+    // Create the arrays
+    bins_.clear();
+    averages_.clear();
+    averages_.resize(bins_.size());
+}
+
+// Zero histogram bins
+void IntegerHistogram1D::zeroBins()
+{
+    bins_.clear();
+    nBinned_ = 0;
+    nMissed_ = 0;
+}
+
+// Set up supplied axis
+void IntegerHistogram1D::setUpAxis(int axisMin, int &axisMax, int binWidth, int &nBins, std::vector<int> &binCentres)
+{
+    // Min, max, and bin width should be set to requested values initially
+    // We will clamp the maximum to the nearest bin boundary (not less than the supplied axisMax)
+    int range = axisMax - axisMin;
+    nBins = range;
+    if ((axisMin + nBins * binWidth) < axisMax)
+    {
+        ++nBins;
+        axisMax = axisMin + nBins * binWidth;
+    }
+}
+
+// Return minimum value for data (hard left-edge of first bin)
+std::optional<int> IntegerHistogram1D::minimum() const { return minimum_; }
+
+// Return maximum value for data (hard right-edge of last bin, adjusted to match bin width if necessary)
+std::optional<int> IntegerHistogram1D::maximum() const { return maximum_; }
+// Bin specified value, returning success
+bool IntegerHistogram1D::bin(int x)
+{
+    auto bin = x;
+    if (bins_.find(bin) == bins_.end())
+        bins_[bin] = std::make_pair(1, averages_);
+    else
+    {
+        bins_[bin].first++;
+        bins_[bin].second = averages_;
+    }
```

Comment:
Couple of things here.  First, the independent `averages_` vector is no longer needed, since we're tying the average value to the bin.  Second, the pair should be `int, SampledDouble` rather than a vector for the latter.  Third, you should be able to simplify the code here by using the `std::map` feature that returns a new value if the accessed key isn't currently present:
```suggestion
bins_[x].first++;
```
Glorious, huh? :smile:

Also need to do the check on min/max limits here before we do the binning.